docs/RFCs/package-toml.md

```diff
@@ -0,0 +1,166 @@
+
+When the `key` value is the qualified package name (e.g. `toddself@static-pkg.dev/package2toml`), then the `value` **MAY** be either a semver range for the package or the <a href="#alias">alias</a> definition.
+
+<a name="alias"></a>
+
+##### Aliases
+
+Aliases can be used to provide short-hand references to longer or unwieldy package names. The only requirement is that they are valid package names and be unique.
+
+They should point to sub-objects which **MAY** contain the following two keys or **MAY** be the short-hand representation of this data. One of these options **MUST** be chosen:
+
+- `"name"` - the fully-qualified name of the package from the registry
+- `"version"` - the semver range of this package you require
+
+Shorthand: `"name@version"`
+
+The object **MAY** also contain:
+
+- `"patch"` - a `key`/`value` pair that allows specific overrides in the dependencies of the dependency. e.g. if this package depends on `lodash`, you could specifiy that you want it to use `underscore` instead: `"patch": {"legacy@npmjs.com/lodash": "toddself@static-pkg.dev/underscore"}`
+
+This section could be used to also specify platform exclusion/inclusion for some dependencies. Some thought might be needed to figure out how to make this workable across the wide range of computer platforms that implementations might be written for.
+
+Examples:
+
+To specify a specific version:
+
+```
+"toddself@static-pkg.dev/package2toml" = "1.0.4"
+```
+
+To alias a package using short-hand syntax:
+
+```
+"package2toml": "toddself@static-pkg.dev/package2toml@^1.0.0"
+```
+
+To alias a package and specify a patch:
+
+```toml
+[dependencies.package2toml]
+"name" = "toddself@static-pkg.dev/package2toml"
+"version" = "^1.0.0"
+[dependencies.package2toml.patch]
+"legacy@npmjs.com/lodash" = "toddself@static-pkg.dev/underscore@1.0.5"
+```
+
+#### Dependency Types
+
+You **MAY** declare dependency types for:
+
+- `[dependencies]` - these are the packages that are required to run this package. They will be installed automatically when this package is required by another package, or when you specifically install the dependencies for this package manually.
+- `[devDependencies]` - these are packages that are required to develop this package. They will be installed only when you install the dependencies for this package manually. If a package depends on the current package, they will **NOT** be installed as part of that dependency graph.
+- `[peerDependencies]` - these packages are rqeuired to run this package, however, they will not be installed and should be additionally depended on or installed by either the parent package or along side this package for development.
+- `[optionalDependencies]` - these packages are not required to run this software, but may provide additional options or features in this package. They must be required or installed by the parent package or along side this package for development.
```

Comment:
I've heard requests for better ways of handling optional dependencies. E.g. how do I not install / compile brotli on newer versions of node.

Reply:
Yeah, i think we have some definite work to do here in the deps section, especially with regard to the notably absent lifecycle scripts...